### DISCUSSION TITLE
OpenGL, zoom, pan and UI restyling

### DISCUSSION THREAD

**Discussion by arkypita** [arkypita/LaserGRBL#2256] [General]
New features:
- GCode preview now is rendered via OpenGL, this allow to zoom and pan the preview
- GCode preview now show engraving progress with different colors
- UI restyling with modern icons, new color scheme, MDI Icons library

This release is pre-release, please report any issue with it.
In menu "Grbl - Settings - Options" it is possible to disable the new features (Legacy Preview and Legacy Icons) 

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/3a899036-86cd-44f8-8cc3-9a657b17ab8e)

<hr /><em>This discussion was created from the release <a href='https://github.com/arkypita/LaserGRBL/releases/tag/v6.9.10'>OpenGL, zoom, pan and UI restyling</a>.</em>

**Comment by ristraus** [comment]
Ciao, Tutto bene?  I sent the *.res files for Portuguese version (PT_BR) some time ago, but I still see many non translated itens, such as Generate menu. Wondering if I'm doing something wrong. Is there an easy why to compile just the *.res files and generate the LaserGRBL.resources.dll in order to test the translation ?

**Reply by arkypita** [reply]
Ciao! Sorry for delay, i miss your pull request, my fault. I'll include in the code asap.

If you want to build LaserGRBL just download and install Visual Studio 2017 community edition and open solution file then press "play" button

**Comment by LogrythmAtGmailDotCom** [comment]
Hi, I love the new update/interface, thanks for your excellent work :)

I notice that the 'digital read out' in the top right of the preview screen (coordinates, power, speed etc) is no longer there. Is there an option to bring it back?

Cheers <3

**Reply by arkypita** [reply]
I plan to restore it, thank you

**Reply by arkypita** [reply]
done: https://github.com/arkypita/LaserGRBL/releases/tag/v6.9.12

**Comment by LogrythmAtGmailDotCom** [comment]
Groovy!!
I really like the fancy new preview window 😎👍

**Comment by Tribo-2dot0** [comment]
Notei duas coisas...
1) Quando se faz alguma mudança que requer reinicialização (como mudança de idioma ou desligar os novos recursos) a antiga janela não é fechada, apenas abre a nova ficando duas instâncias.

2) Em algumas linhas aparece a informação de "Bad number format"; o trabalho não para, segue como se nada houvesse. Fiz o downgrade para a versão 6.2.2 e isso realmente não acontece.
![Captura de tela 2024-05-25 141919](https://github.com/arkypita/LaserGRBL/assets/83982525/53e020d5-cd1e-49a4-aeca-9d07c09b53b7)

**Reply by arkypita** [reply]
Bad number is a message that the controller send back when it does not recognize the g-code as a valid one. Of course LaserGBRL generate valid g-code, but some electrical noise can corrupt communication and make the machine receive wrong codes.

Reboot issue sometime happen also in older version, in my computer too. I should deep this issue (not a major issue, but i like to fix it)

Both are not directly related to the update... but thanks for reporting. You say you don't have this "bad number format" issues in v 6.2.2 so, maybe the opengl thread overload your PC/CPU and it was no longer able to send gcode correctly? this is the only explanation. Did you try to use latest 6.9.10 but disabling opengl preview windows in settings?

**Reply by Tribo-2dot0** [reply]
Esse problema do "bad number format" acontece de ambas maneiras, com o opengl preview ligado ou desligado... nunca havia acontecido em nenhuma das versões anteriores, tanto nesse PC onde uso o Emulador quanto no outro PC ligado às máquinas.
Esse teste com o 6.9.10 foi feito aqui com o Emulador, ainda não testei no outro.

**Reply by arkypita** [reply]
> Esse problema do "bad number format" acontece de ambas maneiras, com o opengl preview ligado ou desligado... nunca havia acontecido em nenhuma das versões anteriores, tanto nesse PC onde uso o Emulador quanto no outro PC ligado às máquinas. Esse teste com o 6.9.10 foi feito aqui com o Emulador, ainda não testei no outro.

Ahahah... my fault! To debug preview windows coloring the executed job (green, or red if errors) i have added some random error response into the Emulator! Lol... i forgot it on!

**Reply by Tribo-2dot0** [reply]
Ainda não está sendo possível definir a Qualidade de Preenchimento de vetorização nos idiomas Português e Húngaro.
![Captura de tela 2024-05-29 133652](https://github.com/arkypita/LaserGRBL/assets/83982525/cefaf679-b5d0-49bc-8308-c1753255bb5f)

**Comment by miab0t** [comment]
I use Ubuntu with WIne, I updated it and now it no longer works. It starts opening and then it closes.

**Reply by arkypita** [reply]
Hi. There is an option to go back to old preview panel (non opengl) but of course if the program crash is it not possible to change this settings. It could be useful to have some info if there is any... by example some error log, message box, or other.

**Reply by arkypita** [reply]
if you run the last version with command line "nogl" it disable the opengl preview https://github.com/arkypita/LaserGRBL/releases/tag/v6.9.12

**Comment by BRUN-Richard** [comment]
Good morning
congratulations for the new version; Zoom really brings a plus!
two small improvements however that would be useful to make:
1°) the display at the top right of the Work and machine coordinates was very useful and has disappeared (Zoom mode)
2°) I often use the Z axis (drill) with movements in G0; Would it be possible to display the movement in Z in the window by a point (red or other) in order to better see the movements (points located on the same line)?
once again Bravo for your fantastic work (I develop it myself and I know what I'm talking about :-))

**Reply by arkypita** [reply]
1. done: https://github.com/arkypita/LaserGRBL/releases/tag/v6.9.12
2. not sure how to do, because my 3d is actually a 2d

**Reply by BRUN-Richard** [reply]
Thanks for your prompt reply 
for the second point, I do not need 3D vision, but simply to be able to visualize the end of a travel in G0; particularly when two G0 travels are aligned
Thanks
Best regards

**Comment by LogrythmAtGmailDotCom** [comment]
Legend!! Thanks, this is great software!
